src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs

```diff
@@ -11,19 +11,33 @@ namespace Microsoft.AspNetCore.Mvc
 {
     public class ObjectResult : ActionResult, IStatusCodeActionResult
     {
+        private MediaTypeCollection _contentTypes;
+
         public ObjectResult(object value)
         {
             Value = value;
             Formatters = new FormatterCollection<IOutputFormatter>();
-            ContentTypes = new MediaTypeCollection();
+            _contentTypes = new MediaTypeCollection();
         }
 
         [ActionResultObjectValue]
         public object Value { get; set; }
 
         public FormatterCollection<IOutputFormatter> Formatters { get; set; }
 
-        public MediaTypeCollection ContentTypes { get; set; }
+        public MediaTypeCollection ContentTypes
+        {
+            get => _contentTypes;
+            set
+            {
+                if (_contentTypes == null)
```

Comment:
`if (value == null)`?

Reply:
oops

Reply:
Pranav points for `_contentTypes = value ?? throw new ArgumentNullException(nameof(value));`?